core/src/main/java/org/apache/iceberg/BaseSnapshot.java

```diff
@@ -126,18 +84,37 @@ class BaseSnapshot implements Snapshot {
       Integer schemaId,
       List<ManifestFile> dataManifests) {
     this(
-        io,
         INITIAL_SEQUENCE_NUMBER,
         snapshotId,
         parentId,
         timestampMillis,
         operation,
         summary,
         schemaId,
-        null);
+        dataManifests.stream().map(ManifestFile::path).toArray(String[]::new));
```

Comment:
Why not set `allManifests` from this list? I think we're probably dropping metadata here, although it doesn't matter much for the tests.